cluster_experiments/power_analysis.py

```diff
@@ -527,6 +547,24 @@ def check_inputs(self):
         self.check_clusters()
 
 
+class PowerAnalysisWithPreExperimentData(PowerAnalysis):
+    """
+    Same as PowerAnalysis, but allowing a perturbation only at experiment period and keeping pre-experiment df intact. The PowerAnalysis class
```

Comment:
I'd mention here that this is supposed to work mainly for diff-in-diff or synthetic control-like estimators.